src/controller/tilemapController.gd

```diff
@@ -1,9 +1,20 @@
 class_name TileMapController extends TileMap
 
+const utils = preload("res://src/utils/matrix_utils.gd")
 
-var grid_state = []
-var patterns_loaded = false
-var cell_pattern_dict = {}
+var old_grid_state: Array = []
```

Comment:
Wir sollten das ganze mal in eine separate Klasse packen, damit wir hier einfach ein dictionary für die unterschiedlichen Botenstoffe reinpacken können. Wir können da mal zusammen drüberschauen.